packages/grafana-ui/src/components/DropZone/FileListItem.tsx

```diff
@@ -0,0 +1,120 @@
+import { css } from '@emotion/css';
+import React from 'react';
+import { GrafanaTheme2 } from '../../../../grafana-data/src';
+import { useStyles2 } from '../../themes';
+import { formatFileSize, trimFileName } from '../../utils/file';
+import { Button } from '../Button';
+import { Icon } from '../Icon/Icon';
+import { IconButton } from '../IconButton/IconButton';
+import { DropzoneFile } from './DropZone';
+
+export interface FileListItemProps {
+  file: DropzoneFile;
+  removeFile?: (file: DropzoneFile) => void;
+}
+
+export function FileListItem({ file: customFile, removeFile }: FileListItemProps) {
+  const styles = useStyles2(getStyles);
+  const { file, progress, error, abortUpload, retryUpload } = customFile;
+
+  const renderRightSide = () => {
+    if (error) {
+      return (
+        <>
+          <span className={styles.error}>{error.message}</span>
+          {retryUpload ? (
+            <IconButton name="sync" tooltip="Retry" tooltipPlacement="top" onClick={() => retryUpload()} />
```

Comment:
```suggestion
            <IconButton name="sync" tooltip="Retry" tooltipPlacement="top" onClick={retryUpload} />
```
No need to create a new function here.